pkg/catalog/catalog.go

```diff
@@ -10,9 +11,140 @@ import (
+const DevfileIndexLink = DevfileRegistry + "/devfiles/index.json"
+
+// GetDevfileIndex loads the devfile registry index.json
+func GetDevfileIndex(devfileIndexLink string) ([]DevfileIndexEntry, error) {
+	var devfileIndex []DevfileIndexEntry
+
+	jsonBytes, err := util.HTTPGetRequest(devfileIndexLink)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to download the devfile index.json from %s", devfileIndexLink)
+	}
+
+	err = json.Unmarshal(jsonBytes, &devfileIndex)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to unmarshal the devfile index.json from %s", devfileIndexLink)
+	}
+
+	return devfileIndex, nil
+}
+
+// GetDevfile loads the devfile
+func GetDevfile(devfileLink string) (Devfile, error) {
+	var devfile Devfile
+
+	yamlBytes, err := util.HTTPGetRequest(devfileLink)
+	if err != nil {
+		return Devfile{}, errors.Wrapf(err, "Unable to download the devfile from %s", devfileLink)
+	}
+
+	err = yaml.Unmarshal(yamlBytes, &devfile)
+	if err != nil {
+		return Devfile{}, errors.Wrapf(err, "Unable to unmarshal the devfile from %s", devfileLink)
+	}
+
+	return devfile, nil
+}
+
+// IsDevfileComponentSupported checks if the devfile is supported
+// The supported devfile should satisfy the following conditions:
+// 1. Devfile has dockerimage as component type
+// 2. Devfile has alias
+// 3. Devfile has run command
+// 4. Devfile has build command
+func IsDevfileComponentSupported(devfile Devfile) bool {
+	hasDockerImage := false
+	hasAlias := false
+	hasRunCommand := false
+	hasBuildCommand := false
+
+	for _, component := range devfile.Components {
+		if hasDockerImage && hasAlias {
+			break
+		}
+
+		if !hasDockerImage {
+			hasDockerImage = strings.Contains(component.Type, "dockerimage")
+		}
+
+		if !hasAlias {
+			hasAlias = len(component.Alias) > 0
+		}
+	}
+
+	for _, command := range devfile.Commands {
+		if hasRunCommand && hasBuildCommand {
+			break
+		}
+
+		if !hasRunCommand {
+			hasRunCommand = strings.Contains(command.Name, "run")
+		}
+
+		if !hasBuildCommand {
+			hasBuildCommand = strings.Contains(command.Name, "build")
+		}
+	}
+
+	if hasDockerImage && hasAlias && hasBuildCommand && hasRunCommand {
+		return true
+	} else {
+		return false
+	}
+}
+
+// ListDevfileComponents lists all the available devfile components
+func ListDevfileComponents() (DevfileComponentTypeList, error) {
+	var catalogDevfileList DevfileComponentTypeList
+	catalogDevfileList.DevfileRegistry = DevfileRegistry
+
+	// Load the devfile registry index.json
+	devfileIndex, err := GetDevfileIndex(DevfileIndexLink)
+	if err != nil {
+		return DevfileComponentTypeList{}, err
+	}
+
+	// 1. Load devfiles that indexed in devfile registry index.json
+	// 2. Populate devfile components with devfile data
+	// 3. Form devfile component list
+	for _, devfileIndexEntry := range devfileIndex {
+		devfileIndexEntryName := devfileIndexEntry.DisplayName
+		devfileIndexEntryDescription := devfileIndexEntry.Description
```

Comment:
Both this and devfileIndexEntryName are uneeded in the code and can be directly incorporated into:
```go
catalogDevfile.Description
```

For example... you don't need this on line 120 and 121..

You can refactor it like so: (see other comment)